internal/testutils/systembus.go

```diff
@@ -0,0 +1,126 @@
+package testutils
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/godbus/dbus/v5"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	sdbus sync.Once
+
+	sdbusMU                sync.Mutex
+	nbRunningTestsSdbus    uint
+	stopDbus               context.CancelFunc
+	dbusCmd                *exec.Cmd
+	config                 string
+	savedDbusSystemAddress string
+)
+
+// StartLocalSystemBus allows to start and set environment variable to a local bus, preventing polluting system ones.
+func StartLocalSystemBus() func() {
+	sdbusMU.Lock()
+	defer sdbusMU.Unlock()
+	nbRunningTestsSdbus++
+
+	sdbus.Do(func() {
+		dir, err := os.MkdirTemp("", "tests-dbus")
+		if err != nil {
+			log.Fatalf("Setup: can’t create dbus system directory: %v", err)
+		}
+
+		savedDbusSystemAddress = os.Getenv("DBUS_SYSTEM_BUS_ADDRESS")
+		config = filepath.Join(dir, "dbus.config")
+		err = os.WriteFile(config, []byte(`<!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-Bus Bus Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
+<busconfig>
+  <type>system</type>
+  <keep_umask/>
+  <listen>unix:tmpdir=/tmp</listen>
+  <standard_system_servicedirs />
+  <policy context="default">
+    <allow send_destination="*" eavesdrop="true"/>
+    <allow eavesdrop="true"/>
+    <allow own="*"/>
+  </policy>
+</busconfig>`), 0600)
+		if err != nil {
+			log.Fatalf("Setup: can’t create dbus configuration: %v", err)
+		}
+		var ctx context.Context
+		ctx, stopDbus = context.WithCancel(context.Background())
+		// #nosec G204 - this is only for tests, we are in control of the config
+		dbusCmd = exec.CommandContext(ctx, "dbus-daemon", "--print-address=1", "--config-file="+config)
+		dbusStdout, err := dbusCmd.StdoutPipe()
+		if err != nil {
+			_ = os.RemoveAll(dir)
+			log.Fatalf("couldn't get stdout of dbus-daemon: %v", err)
+		}
+		if err := dbusCmd.Start(); err != nil {
+			_ = os.RemoveAll(dir)
+			log.Fatalf("couldn't start dbus-daemon: %v", err)
+		}
+		dbusAddr := make([]byte, 256)
+		n, err := dbusStdout.Read(dbusAddr)
+		if err != nil {
+			_ = os.RemoveAll(dir)
```

Comment:
If we are ignoring the value, can't it be just something like `os.RemoveAll(dir)`?